### DISCUSSION TITLE
Merge several files using v4

### DISCUSSION THREAD

**Discussion by luisR2** [mikefarah/yq#688] [Q&A] [answered]
Hi,

I was using the v3 version to merge several yaml files (inside the folder configFiles) into one combined file using this commad:

> yq m -a=append ./configFiles/*.yml > ./newConfig/combined.yml

Now I'm trying to get the same result with v4 but I'm not getting the same combined file. The command that I'm using is:

> yq eval-all 'select(fileIndex == 0) * select(filename == "*.yml")' ./configFiles/*.yml > ./newConfig/combined.yml

Which would be the equivalent v4 command as the previous v3 that I was using?

Thanks!

**Comment by mikefarah** [comment]
This should do it:
https://mikefarah.gitbook.io/yq/operators/multiply#merge-appending-arrays

**Comment by luisR2** [comment]
Thanks for the answer, I tried the "*+"  but didnt' work. The idea is to merge f2.yml not into f1.yml but in a new created yml (https://mikefarah.gitbook.io/yq/commands/evaluate-all) like it was done in the pass with:
`yq merge -a=true /path/to/yaml/files `

I mean maybe it can't be done anymore in this new v4 version?

**Comment by mikefarah** [comment, marked as answer]
Ah ok - this looks like it's related to this enhancement request: https://github.com/mikefarah/yq/issues/674 - just need to figure out how to actually implement it (and get the time to do it :S )

**Reply by luisR2** [reply]
Hey you are right, that request is exactly what I need. Well it will be great if that is implemented someday. I will then remain in v3 until this feature is released 

Thanks so much!

**Reply by mikefarah** [reply]
You can now do this in 4.6.0:

https://mikefarah.gitbook.io/yq/operators/reduce#merge-all-yaml-files-together

**Reply by itoffshore** [reply]
this works well for merging nodes together (extra `+`):

`yq eval-all '. as $item ireduce ({}; . *+ $item )' file.yaml file2.yml`

**Comment by luisR2** [comment]
Looking forward to check this new version, thanks for notifying!
